File: rb/features.rb
# ChineseMedicineClinics SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ChineseMedicineClinicsFeatures
  def self.make_feature(name)
    case name
    when "base"
      ChineseMedicineClinicsBaseFeature.new
    when "ratelimit"
      ChineseMedicineClinicsRatelimitFeature.new
    when "retry"
      ChineseMedicineClinicsRetryFeature.new
    when "test"
      ChineseMedicineClinicsTestFeature.new
    when "timeout"
      ChineseMedicineClinicsTimeoutFeature.new
    else
      ChineseMedicineClinicsBaseFeature.new
    end
  end
end

File: py/chinesemedicineclinics_sdk/features.py
# ChineseMedicineClinics SDK feature factory

from chinesemedicineclinics_sdk.feature.base_feature import ChineseMedicineClinicsBaseFeature
from chinesemedicineclinics_sdk.feature.test_feature import ChineseMedicineClinicsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ChineseMedicineClinicsBaseFeature(),
        "test": lambda: ChineseMedicineClinicsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
